src/jaxsim/rbda/contacts/relaxed_rigid.py

```diff
@@ -0,0 +1,356 @@
+
+import jaxsim.api as js
+import jaxsim.typing as jtp
+from jaxsim.api.common import VelRepr
+from jaxsim.math import Adjoint
+from jaxsim.terrain.terrain import FlatTerrain, Terrain
+
+from .common import ContactModel, ContactsParams, ContactsState
+
+
+@jax_dataclasses.pytree_dataclass
+class RelaxedRigidContactsParams(ContactsParams):
+    """Parameters of the relaxed rigid contacts model."""
+
+    # Time constant
+    time_constant: jtp.Float = dataclasses.field(
+        default_factory=lambda: jnp.array(0.01, dtype=float)
+    )
+
+    # Adimensional damping coefficient
+    damping_coefficient: jtp.Float = dataclasses.field(
+        default_factory=lambda: jnp.array(1.0, dtype=float)
+    )
+
+    # Minimum impedance
+    d_min: jtp.Float = dataclasses.field(
+        default_factory=lambda: jnp.array(0.9, dtype=float)
+    )
+
+    # Maximum impedance
+    d_max: jtp.Float = dataclasses.field(
+        default_factory=lambda: jnp.array(0.95, dtype=float)
+    )
+
+    # Width
+    width: jtp.Float = dataclasses.field(
+        default_factory=lambda: jnp.array(0.0001, dtype=float)
+    )
+
+    # Midpoint
+    midpoint: jtp.Float = dataclasses.field(
+        default_factory=lambda: jnp.array(0.1, dtype=float)
+    )
+
+    # Power exponent
+    power: jtp.Float = dataclasses.field(
+        default_factory=lambda: jnp.array(1.0, dtype=float)
+    )
+
+    # Stiffness
+    stiffness: jtp.Float = dataclasses.field(
+        default_factory=lambda: jnp.array(0.0, dtype=float)
+    )
+
+    # Damping
+    damping: jtp.Float = dataclasses.field(
+        default_factory=lambda: jnp.array(0.0, dtype=float)
+    )
+
+    # Friction coefficient
+    mu: jtp.Float = dataclasses.field(
+        default_factory=lambda: jnp.array(0.5, dtype=float)
+    )
+
+    # Maximum number of iterations
+    max_iterations: jtp.Int = dataclasses.field(
+        default_factory=lambda: jnp.array(50, dtype=int)
+    )
+
+    # Solver Tolerance
+    tolerance: jtp.Float = dataclasses.field(
+        default_factory=lambda: jnp.array(1e-6, dtype=float)
+    )
+
+    def __hash__(self) -> int:
+        from jaxsim.utils.wrappers import HashedNumpyArray
+
+        return hash(
+            (
+                HashedNumpyArray(self.time_constant),
+                HashedNumpyArray(self.damping_coefficient),
+                HashedNumpyArray(self.d_min),
+                HashedNumpyArray(self.d_max),
+                HashedNumpyArray(self.width),
+                HashedNumpyArray(self.midpoint),
+                HashedNumpyArray(self.power),
+                HashedNumpyArray(self.stiffness),
+                HashedNumpyArray(self.damping),
+                HashedNumpyArray(self.mu),
+                HashedNumpyArray(self.max_iterations),
+                HashedNumpyArray(self.tolerance),
+            )
+        )
+
+    def __eq__(self, other: RelaxedRigidContactsParams) -> bool:
+        return hash(self) == hash(other)
+
+    @classmethod
+    def build(
```

Comment:
Can you use all `*Like` types here? Then, when building the class, cast all non-static fields to JAX arrays.

Reply:
Done in https://github.com/ami-iit/jaxsim/pull/223/commits/89843345142c7d73b88be31940d376cf947bcf38 and https://github.com/ami-iit/jaxsim/pull/223/commits/0599e59b63ca628dca5cbb6da8852a7939673fc8